### PR TITLE
feat(replays): add segmented control to sidebar for npm/jsLoader instructions

### PR DESCRIPTION
**Pull request by michellewzhang**
testing with only the `javascript-react` platform for now since https://github.com/getsentry/sentry/pull/61448 & a few other platform conversions still needs to be merged.

add a segmented control to the replay onboarding flyout sidebar, to determine whether the NPM or JS Loader instructions should be shown.

https://github.com/getsentry/sentry/assets/56095982/531e2891-b339-4f22-bc15-494aedbe4397

relates to https://github.com/getsentry/sentry/issues/61069

note: one thing we haven't accounted for yet here: whether the user has "Enable session replay" checked inside the project settings -> DSN -> Loader page

image

### CODE REVIEW

File: static/app/gettingStartedDocs/javascript/jsLoader/react.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+type Params = DocsParams;
+
+const getInstallConfig = (params: Params) => [
+  {
+    type: StepType.INSTALL,
+    configurations: [
+      {
+        description: t('Add this script tag to the top of the page:'),
+        language: 'html',
+        code: `<script src="${params.cdn}" crossorigin="anonymous"></script>`,
```
Comment (michellewzhang) [inline static/app/gettingStartedDocs/javascript/jsLoader/react.tsx:21]: this code snippet could probably be extracted and re-used in the future

File: static/app/components/onboarding/gettingStartedDoc/onboardingLayout.tsx [hunk 2]
```diff
@@ -38,6 +38,7 @@ export type OnboardingLayoutProps = {
   projectId: Project['id'];
   projectSlug: Project['slug'];
   activeProductSelection?: ProductSolution[];
+  cdn?: string;
```
Comment (bruno-garcia) [inline static/app/components/onboarding/gettingStartedDoc/onboardingLayout.tsx:41]: I see references to Loader below, is is the CDN or the Loader link?

Loader is custom for the user. CDN is a static JS file:

CDN (now a heading in the Loader page): https://docs.sentry.io/platforms/javascript/install/loader/#cdn
Loader: https://docs.sentry.io/platforms/javascript/install/loader/#using-the-loader

**Review by ryan953 — APPROVED** [review]

File: static/app/components/replaysOnboarding/sidebar.tsx [hunk 3]
```diff
@@ -229,14 +264,18 @@ function OnboardingContent({currentProject}: {currentProject: Project}) {
           {platform: currentPlatform?.name || currentProject.slug}
         )}
       </IntroText>
-      {newOnboarding && migrated.includes(currentPlatform.id) ? (
+      {showNewOnboardingUI ? (
         <SdkDocumentation
```
Comment (ArthurKnaus) [inline static/app/components/replaysOnboarding/sidebar.tsx:268]: general nit: Not sure if we should use the `SdkDocumentation` component here 🤔
I would rather prefer a solution where we have a base component that does not implement UI and logic which is specific to the general onboarding use case.

**Review by ArthurKnaus — APPROVED** [review]
Like it 👍

File: static/app/components/replaysOnboarding/sidebar.tsx (hunk 3 above)
Comment (michellewzhang) [reply static/app/components/replaysOnboarding/sidebar.tsx:268]: after discussion, we've decided the structure is fine as is and can be refactored later

**Pull request merged**